scripts/linters/pylint_extensions.py

```diff
@@ -2139,6 +2141,54 @@ def visit_classdef(self, node):
             self.add_message('no-dunder-metaclass', node=node)
 
 
+class StringConcatenationChecker(checkers.BaseChecker):
+    """Custom pylint checker which prohibits use of string concatenation and
+     promotes string interpolation instead.
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+
+    name = 'no-string-concatenation'
+    priority = -1
+    msgs = {
+        'C0035': (
+            'at %s, avoid using string concatenation and replace it with '
+            'string '
+            'interpolation instead.',
+            'no-string-concatenation',
+            'Enforce use of string interpolation over string concatenation.',
+        ),
+    }
+
+    def is_node_string_constant(self, node):
+        """Helper function for visit_binop. Called to check if any of the
+        operand is a string constant.
+        """
+        if (isinstance(node, astroid.nodes.Const) and
+                ('str' in node.pytype() or
+                 'unicode' in node.pytype())):
+            return True
+        return False
+
+    def visit_binop(self, node):
+        """Called for every '+' operator to prohibit usage of string
+         concatenation. It covers only those cases where at least one of the
+         operands is astroid.nodes.Const.
+
+        Args:
+            node: astroid.node.BinOp. Node to access module content.
+        """
+        if node.op != b'+':
+            return
+        for operand in node.get_children():
+            if self.is_node_string_constant(operand):
```

Comment:
For the string where the user has tried concatenating two strings should we ask to remove the concatenation as it's not needed? (Currently, we are asking them to use interpolation for such cases.)

Reply:
Yes, I think we can show a separate for that. 
Great idea!! Thanks